pandas/core/arrays/categorical.py

```diff
@@ -746,6 +746,15 @@ def categories(self):
 
     @categories.setter
     def categories(self, categories):
```

Comment:
cna you instead make a _categories and use that internally?

Reply:
we _really_ shouldn't be modifying dtypes inplace at all.  absolute footgun.